src/main/java/org/mitre/dsmiley/httpproxy/URITemplateProxyServlet.java

```diff
@@ -55,7 +57,7 @@
   * But that's not how the spec works. So for now we will require a proxy arg to be present
   * if defined for this proxy URL.
   */
-  protected static final Pattern TEMPLATE_PATTERN = Pattern.compile("\\{([a-zA-Z0-9_%.]+)\\}");
+  protected static final Pattern TEMPLATE_PATTERN = Pattern.compile("\\{([a-zA-Z0-9_%-.]+)\\}");
```

Comment:
I was just looking at the RFC spec.  It turns out most characters are supported except a small number.  To keep this simple excepting all the characters we should (yet some we shouldn't admittedly), lets simplify this: `Pattern.compile("\\{(.+?)\\}")`